File: DataStructure/54_maximumSubarray/main.cpp
#include <bits/stdc++.h>
using namespace std;
int maxSubArray(vector<int> &nums)
{
    int sum = nums[0];
    int maxSize = nums[0];
    for (int i = 1; i < nums.size(); i++)
    {
        sum = max(sum + nums[i], nums[i]);
        maxSize = max(sum, maxSize);
    }
    return maxSize;
}
int main()
{
    vector<int> a;
    a.push_back(-2);
    a.push_back(3);
    a.push_back(-4);

    cout << maxSubArray(a) << endl;
    return 0;
}

File: DataStructure/54_maximumSubarray/main.py
from pip import List


def maxSubArray(self, nums: List[int]) -> int:
    sum = nums[0]
    max = nums[0]
    for i in range(len(nums)):
        if i == 0:
            continue
        sum += nums[i]
        if sum < nums[i]:
            sum = nums[i]
        if max < sum:
            max = sum
    return max


nums = [1]
ans = maxSubArray(0, nums)
print(ans)
